Add codespell support (config, workflow to detect/not fix) and make it fix few typos

yarikoptic opened the pull request:
More about codespell: https://github.com/codespell-project/codespell .

I personally introduced it to dozens if not hundreds of projects already and so far only positive feedback.

CI workflow has 'permissions' set only to 'read' so also should be safe.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/gohugoio/hugo?pullRequest=12731) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/gohugoio/hugo?pullRequest=12731) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/gohugoio/hugo?pullRequest=12731) it.</sub>

yarikoptic commented:
meanwhile I have signed SLA, rebased, redid codespell fixes with newer codespell (to address conflict + account new found on CI but not locally typo), and fixed indentation after codespell fix which lead to some linting CI jobs to fail.

bep commented:
You need to include the config setup ONLY.

bep commented:
I'm closing this for now. This may be interesting some time in the future, but as for now it creates too much noise.
